sendbuffer.go

```diff
@@ -112,36 +131,47 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 }
 
 func (buf *sendBuffer) send(b []byte, writeDeadline time.Time) (int, error) {
-	buf.muClosing.RLock()
-	n, err := buf.doSend(b, writeDeadline)
-	buf.muClosing.RUnlock()
-	return n, err
+	for {
```

Comment:
Need to do this trick here so that we don't deadlock on trying to write to `buf.in` while also trying to close it.